src/Compilers/Core/Portable/Operations/InstanceReferenceKind.cs

```diff
@@ -15,6 +15,10 @@ public enum InstanceReferenceKind
         /// <summary>
         /// Reference to the object being initialized in a C# or VB object or collection initializer, or to the object being referred to in a VB With statement.
         /// </summary>
-        ImplicitReceiver
+        ImplicitReceiver,
+        /// <summary>
+        /// Reference to the anonymous object being initialized in a C# or VB anonymous type object initializer, or to the object being referred to in a VB With statement.
+        /// </summary>
+        ImplicitAnonymousReceiver,
```

Comment:
Why not just use ImplicitReceiver for this? If we can't use ImplicitReceiver, we should rename ImplicitReceiver so that these names don't overlap (since the API hasn't shipped yet, this is our last chance)

Reply:
We can discuss this at today's meeting - I thought we wanted more richer information to distinguish without going back to syntax.

---
In reply to: [196878486](https://github.com/dotnet/roslyn/pull/28005#discussion_r196878486) [](ancestors = 196878486)

Reply:
> We can discuss this at today's meeting - I thought we wanted more richer information to distinguish without going back to syntax.

I think we already using it for different things like With statement as well, for example. 

---
In reply to: [196897129](https://github.com/dotnet/roslyn/pull/28005#discussion_r196897129) [](ancestors = 196897129,196878486)

Reply:
Alright, I have switched back to using the existing field and updated the documentation. I also realized that distinguishing this case does not need to go back to syntax. The client can check `instanceReferenceOperation.Type.IsAnonymousType`.

---
In reply to: [196902409](https://github.com/dotnet/roslyn/pull/28005#discussion_r196902409) [](ancestors = 196902409,196897129,196878486)

Reply:
> The client can check instanceReferenceOperation.Type.IsAnonymousType.

Relying on that would be error prone, an instance of an anonymous type can be a target of a With statement for example. But you are correct, one doesn't need to go back to syntax. It  will be sufficient to go up the parents chain until a node that can create a context like that is reached.

---
In reply to: [197145865](https://github.com/dotnet/roslyn/pull/28005#discussion_r197145865) [](ancestors = 197145865,196902409,196897129,196878486)